### PR TITLE
(feat) O3-3333: Include telephone number in excel sheet export 

### PR DESCRIPTION
**Pull request by lucyjemutai**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

https://openmrs.atlassian.net/browse/O3-3333
Telephone Number should be included in Download,
## Screenshots

image

### CODE REVIEW

**Review by mogoodrich — COMMENTED** [review]
This looks like the correct way to do this, but I think the problem is that the endpoint that the Appointments module returns is not a FHIR Patient and does not have the 'telecom" property?

**Comment by lucyjemutai** [comment]
@ojwanganto That makes sense.  Excluding phone numbers by default helps safeguard patient data.

**Comment by lucyjemutai** [comment]
Thank you @mogoodrich . Should we return `-- `or `null` when a patient has no phone number?

**Comment by mogoodrich** [comment]
> Thank you @mogoodrich . Should we return `-- `or `null` when a patient has no phone number?

I don't have a strong preference, but I would think we just keep the cell empty.

File: packages/esm-appointments-app/src/config-schema.ts [hunk 1]
```diff
@@ -2,6 +2,11 @@ import { Type, restBaseUrl, validators } from '@openmrs/esm-framework';
 import { spaHomePage } from './constants';
 
 export const configSchema = {
+  includeTelephoneNumbers: {
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/config-schema.ts:5]:
```suggestion
  includeTelephoneNumbersInExcelDownload: {
```

File: packages/esm-appointments-app/src/config-schema.ts [hunk 2]
```diff
@@ -146,4 +151,5 @@ export interface ConfigObject {
   showUnscheduledAppointmentsTab: boolean;
   useBahmniAppointmentsUI: boolean;
   useFullViewPrivilege: boolean;
+  includeTelephoneNumbers: boolean;
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/config-schema.ts:154]:
```suggestion
  includeTelephoneNumbersInExcelDownload: boolean;
```

File: packages/esm-appointments-app/src/helpers/excel.ts [hunk 3]
```diff
@@ -1,21 +1,52 @@
-import { formatDate } from '@openmrs/esm-framework';
+import { formatDate, openmrsFetch, fhirBaseUrl, getConfig, useConfig } from '@openmrs/esm-framework';
 import * as XLSX from 'xlsx';
 import { type Appointment } from '../types';
+import { type ConfigObject } from '../config-schema';
+import { moduleName } from '../constants';
+
+async function fetchPatient(patientUuid: string): Promise<fhir.Patient> {
```
[the remark this answers is not in the document]
Comment (denniskigen) [reply packages/esm-appointments-app/src/helpers/excel.ts:7]: [fetchCurrentPatient](https://github.com/openmrs/openmrs-esm-core/blob/342781bbc5148896d028994253a818d462451652/packages/framework/esm-api/src/shared-api-objects/current-patient.ts#L22) should do the trick.

File: packages/esm-appointments-app/src/helpers/excel.ts [hunk 4]
```diff
@@ -1,21 +1,54 @@
...
-    'Patient name': appointment.patient.name,
-    Gender: appointment.patient.gender === 'F' ? 'Female' : 'Male',
-    Age: appointment.patient.age,
-    Identifier: appointment.patient.identifier ?? '--',
-    'Appointment type': appointment.service?.name,
-    Date: formatDate(new Date(appointment.startDateTime), { mode: 'wide' }),
-  }));
+export async function downloadAppointmentsAsExcel(appointments: Array<Appointment>, fileName = 'Appointments') {
+  const config = await getConfig<ConfigObject>(moduleName);
+  const INCLUDE_TELEPHONE_NUMBERS = config.includeTelephoneNumbers ?? false;
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/helpers/excel.ts:21]:
```suggestion
  const includeTelephoneNumbers = config.includeTelephoneNumbersInExcelDownload ?? false;
```

File: packages/esm-appointments-app/src/helpers/excel.ts [hunk 5]
```diff
@@ -1,21 +1,54 @@
...
+
+  const appointmentsJSON = await Promise.all(
+    appointments.map(async (appointment: Appointment) => {
+      const patientinfo = await fetchPatient(appointment.patient.uuid);
+      const phoneNumber =
+        INCLUDE_TELEPHONE_NUMBERS && patientinfo?.telecom
+          ? patientinfo.telecom.map((telecomObj) => telecomObj?.value).join(', ')
+          : '';
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/helpers/excel.ts:29]:
```suggestion
      const patientInfo = await fetchPatient(appointment.patient.uuid);
      const phoneNumber =
        includeTelephoneNumbers && patientInfo?.telecom
          ? patientinfo.telecom.map((telecomObj) => telecomObj?.value).join(', ')
          : '';
```

File: packages/esm-appointments-app/src/helpers/excel.ts [hunk 6]
```diff
@@ -1,21 +1,54 @@
...
+  );
+
+  const headers = [
+    'Patient name',
+    'Gender',
+    'Age',
+    'Identifier',
+    'Appointment type',
+    'Date',
+    ...(INCLUDE_TELEPHONE_NUMBERS ? ['Telephone number'] : []),
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/helpers/excel.ts:50]:
```suggestion
    ...(includeTelephoneNumbers ? ['Telephone number'] : []),
```

**Review by denniskigen — APPROVED** [review]
Mostly LGTM. I've left some suggestions.

**Comment by lucyjemutai** [comment]
Thank you @denniskigen

**Review by ojwanganto — APPROVED** [review]

**Review by CynthiaKamau — APPROVED** [review]

**Review by denniskigen — APPROVED** [review]

**Pull request merged**